src/threading.c

```diff
@@ -1,11 +1,13 @@
 // This file is a part of Julia. License is MIT: https://julialang.org/license
-
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <inttypes.h>
-
+#ifndef _OS_WINDOWS_
+#include <dlfcn.h>
+#endif
+#include <libgen.h> // defines dirname
 #include "julia.h"
 #include "julia_internal.h"
 #include "julia_assert.h"
```

Comment:
```suggestion
#include "julia.h"
#include "julia_internal.h"
#include "julia_assert.h"
#ifndef _OS_WINDOWS_
#include <dlfcn.h>
#endif
#include <libgen.h> // defines dirname
```

`_OS_WINDOWS_` is apparently provided by system headers